floodlight/models/geometry.py

```diff
@@ -209,3 +199,121 @@ def stretch_index(self, xy: XY, axis: str = None) -> TeamProperty:
+    containing all outfield players [reference].
+
+    The convex hull is also known in the literature under the terms ‘surface area’,
+    ‘coverage area’ and ‘playing area’ [1].
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from floodlight import XY
+    >>> from floodlight.models.geometry import ConvexHullModel
+
+    >>> xy = XY(np.array(   [0, 0, 10, 0, 10, 10, 0, 10, 5, 5], # square (10x10)
+    >>>                     [0, 0, 10, 0, 5, 10, 2, 2, 3, 3],   # triangle ((1/2)x10x10)
+    >>>                     [0, 5, 5, 10, 10, 5, 5, 0, 5, 5]))  # rhombus (5x10)
+    >>> exclude_xIDs = [4]
+    >>> chull = ConvexHullModel()
+    >>> chull.fit(xy=xy, exclude_xIDs=exclude_xIDs)
+    >>> chull.area_convex_hull()
+    TeamProperty(property=array([100, 50, 50]), name='area_convex_hull', framerate=None)
+
+    The area enclosed by the convex hull can be visualised using matplotlib as follows:
+
+    >>> import matplotlib.pyplot as plt
+    >>> c_hulls=[]
+    >>> c_hulls = chull.convex_hulls
+    >>> plt.plot(c_hulls[0].points[:, 0], c_hulls[0].points[:, 1], 'o')
+    >>> for simplex in c_hulls[0].simplices:
+    >>>     plt.plot(c_hulls[0].points[simplex, 0], c_hulls[0].points[simplex, 1], 'k-')
+    >>> plt.show()
+
+    .. image:: ../../_img/convex_hull.png
+
+    References
+    ----------
+        .. [1] `Low, B., Coutinho, D., Gonçalves, B., Rein, R., Memmert, D., & Sampaio,
+            J. (2020). A Systematic Review of Collective Tactical Behaviours in Football
+            Using Positional Data. Sports Medicine, 50(2), 343–385.
+            <https://link.springer.com/article/10.1007/s40279-019-01194-7>`_
+    """
+
+    def __init__(self):
+        super().__init__()
+        # model parameter
+        self.convex_hulls = None
+
+    def fit(self, xy: XY, exclude_xIDs: list = None, **kwargs):
+        """
+        Fit the model to the given data and create ConvexHull objects from scipy.spatial
+        for each frame.
+
+        Parameters
+        ----------
+        xy: XY
+            Player spatiotemporal data for which the ConvexHull object is created.
+        exclude_xIDs: list, optional
+            A list of xIDs to be excluded from computation. This can be useful if one
+            would like, for example, to exclude goalkeepers from analysis.
+        kwargs:
+            Optional keyworded arguments e.g. {'incremental: bool = False',
+            'qhull_options: str' = None} which are passed to the ConvexHull objects
+            (`scipy.spatial.ConvexHull
+            <https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.ConvexHull.
+            html>`_).
+        """
+
+        # array to indicate which players are to be excluded from the calculation
+        include = _exclude_x_ids(xy, exclude_xIDs)
+
+        # check for NaN's in xy since ConvexHull from scipy can not handle NaN values
+        if np.isnan(xy[:, include]).any():
+            raise ValueError("xy array contains NaN")
```

Comment:
that's not an ideal nan-handling as it shuts down the whole thing if there is only a single nan in the whole array. as the hulls are temporally independent anyway, I think it would be more appropriate to check this within the for loop and compute hulls for valid timepoints.

additionally, there will most likely be nans in "good" timepoints, too, e.g. due to subbed players. ideally, it would then still be possible to calculate the hull with the remaining players, but I assume this cries for a more sophisticated nan-handling outside this classes' scope